src/Sylius/Bundle/ThemeBundle/DependencyInjection/Compiler/ThemeCompilerPass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Sylius\Bundle\ThemeBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Resource\FileResource;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+class ThemeCompilerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $definition = $container->findDefinition('sylius.theme.repository');
+
+        $loader = $container->get('sylius.theme.loader');
+
+        $serializedThemes = [];
+        try {
+            $themeFiles = $container->get('sylius.theme.locator')->locate('theme.json', null, false);
+            foreach ($themeFiles as $themeFile) {
+                $serializedThemes[] = serialize($loader->load($themeFile));
+                $container->addResource(new FileResource($themeFile));
+            }
+
+            $definition->addArgument($serializedThemes);
```

Comment:
Ok, for me method ``process`` is quite general and does not say almost anything about functionality, so i would precise it in private methods - but i know it can be controversial for some, so I'm not persuading. However i still think i would be nicer ;)